python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -801,18 +755,25 @@ def _process_auto_materialize_tick_generator(
 
             # Determine if the most recent tick requires retrying
             retry_tick: Optional[InstigatorTick] = None
-
+            override_evaluation_id: Optional[int] = None
             if latest_tick:
+                can_resume = (
+                    get_current_timestamp() - latest_tick.timestamp
+                ) <= MAX_TIME_TO_RESUME_TICK_SECONDS
+                previous_cursor_written = (
+                    latest_tick.tick_data.auto_materialize_evaluation_id
+                    == stored_cursor.evaluation_id
+                )
+                if can_resume and not previous_cursor_written:
+                    # if the tick failed before writing a cursor, we don't want to advance the
+                    # evaluation id yet
+                    override_evaluation_id = latest_tick.tick_data.auto_materialize_evaluation_id
```

Comment:
do we have a specific test that would fail if this was not working properly

Reply:
sort of unfortunate that these will always match except in this somewhat unusual edge case, i am a little worried that some code somewhere will assume that they always match

Reply:
several tests actually, this was kinda painful to get passing all the"make sure we handle tick failures properly" cases!